worker.go

```diff
@@ -149,6 +149,45 @@ func (l *loadBalancer) Owner() string {
 	return ""
 }
 
+// CertificatesFinder interface represents a list of certificates
+// and some basic operations than can be performed on them.
+type CertificatesFinder interface {
```

Comment:
@marcinzaremba since you refactor this, would it make sense to move it out of `worker.go` to a separate file? `worker.go` is already a bit bloated with stuff :)

Reply:
I was thinking about this but cannot come up with meaningful separate component (area of responsibility. And I would like not to end up with something pretty meaningless like `worker_utils.go`. Can you maybe suggest some possibilities?

Reply:
`certificates_finder.go` would be fine for me, it doesn't have to be in a separate package, just to things out of `worker.go`. If you don't want to do it in this PR, it's also totally fine, I just wanted to suggest it :)